components/ProductCard.js

```diff
@@ -0,0 +1,117 @@
+import styled from 'styled-components';
+import QuantitySelector from './QuantitySelector';
+import { useState } from 'react';
+
+export default function ProductCard({ product }) {
+  const [showDetails, setShowDetails] = useState();
+
+  return (
+    <>
+      <StyledCard key={product.id}>
+        <StyledImage src={product.image} alt={product.name} />
+        <MainInfoWrapper>
+          <StyledProductName>{product.name}</StyledProductName>
+          <StyledWSPrice>WS {product.WSprice.toPrecision(4)} €</StyledWSPrice>
```

Comment:
take a look at `toLocaleString`
there's an option for currency
https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString

Reply:
Nice! Thank you!